charts/tidb-lightning/templates/scripts/_start_lightning.sh.tpl

```diff
@@ -0,0 +1,22 @@
+data_dir=$(dirname $(find /var/lib/tidb-lightning -name metadata 2>/dev/null) 2>/dev/null)
+if [ -z $data_dir]; then
+    echo "No mydumper files are found, please exec into my container to diagnose"
+    tail -f /dev/null
```

Comment:
The echo should be to stderr `>&2`.

Instead of holding this container open by default, maybe we can have an environment variable or some kind of debug switch to turn that behavior on?

Reply:
The backup data is usually very large, it's costly to restart from scratch.
With the container open, users can exec into the container to diagnose and run the tidb-lightning manually.

Reply:
Yes, but how will we know when it fails?

For large data in the cloud we need to stream the data directly from a storage bucket (but download the metadata first).

Reply:
I don't think tidb-lightning supports streaming data from cloud storage right now.

Reply:
I've added an option to allow fail fast, the job status will indicate the restore status when `failFast` is enabled.

Reply:
failFast is good, but still how will someone know that lightning has failed? It seems like the only thing they can do is tail the log.
And then how does one fix the situation?
If we store all data in PVC that are retained, then shouldn't it be easy to launch a new job with a shell for the user to complete the process?

Reply:
If `failFast` is enabled, the job fails means the lightning fails, no need to tail the log (After https://github.com/pingcap/tidb-lightning/pull/230 is merged).
The recommended workflow is:
1. Deploy tidb-lightning with `failFast` enabled.
2. If the job fails (indicated by job status), delete the failed job with `kubectl delete job -n <namespace> <release-name-tidb-lightning>`
3. Use helm and kubectl to launch the job with `failFast` disabled by `helm template pingcap/tidb-lightning --name <release-name> | kubectl apply -n <namespace> -f -`. This will keep the pod running when fails.

Reply:
I leave the default `failFast` to `false` to make it handy for users to handle when there are errors.
If the first try fails, the second running should quickly get the debug shell. The failure should happen instantly. So no need to add "debug statements and then sleep", the current setup should be enough.